### PR TITLE
fix(es/compat): Visit transformed expression in optional chaining pass

### LINKED ISSUE #7612: optional chaining syntax is not compiled
### Describe the bug
When there are multiple optional chaining syntaxes, some will be compiled successfully, but some will not be compiled.

image

### Input code

```typescript
const ModalInner = ({ data }: { data: GetReturnOrderDetails.Result }) => {
  return (
    <div className={styles.qcWrapper} ref={wrapperRef} id="qcDataScroll">
      {data!.qcDataList.map(qcData => (
        <div className={styles.desc} key={qcData.flawDesc}>
          <div className={styles.descTitle}>{qcData?.flawDesc}</div>
          {qcData?.flawImageList?.length === 0 ? (
            <Empty />
          ) : (
            <Album
              data={(qcData?.flawImageList || [])?.map(src => ({
                url: src,
                describe: qcData?.flawDesc
              }))}
              showIndicators
              useSafeAreaBottom
              showDescription
              current={current}
            >
              test
            </Album>
          )}
        </div>
      ))}
    </div>
  )
}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true
    },
    "target": "es5",
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    }
  },
  "module": {
    "type": "es6"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link

https://play.swc.rs/?version=1.3.66&code=H4sIAAAAAAAAA31Ry27CMBC85yu2nBKpCj1HOIiKqkLqQ4JKPVQ9GGeBqM4De1OEjP%2B9eVAg7uO23pnZ2R17osg1wWORcDnLc1TAwDeQcOJgI%2BiqCO6R5kiVyp9VgmqKxFOpwznqShLYAFgMxgNQLQf8ugQYJeknCMm1fuIZMqNpL1GHW%2FGqeFmisjV9xcyue81xZSFN2GArprXlQqhCykHcTgIwzRpXYYc9pJrCjJd%2B92zM%2FSPvT9cEtbDwgXtmOlW4knw3bbrxSfuv%2BiUliTY%2Bysdn%2FWhYiy6H9CizjK%2Bx2XgcSszXtAHGGNzA%2BGLn1vkuK2kPw8tBAUQuayKXVdZrQftFzPi%2FusLhAG%2FvwbjNSyvRhmWcAQCVkhHU8PUPpDldpUuMwD3codogsE5Lb4rdLE9SwalQ2gErjQu%2BwolCflsQFe5VjXjampeUFrmDikopzImZY9G3jh02oaZ%2BjsM2yF7Y5xG9H%2F0%2B69QMPOt5X%2B8ckCE4AwAA&config=H4sIAAAAAAAAA1WPSw7CMAxE9z1F5DVbWHAHDhEFtwrKT7ErEVW9O%2FkDu%2FjNTMY%2BFiHgRQru4sjPPAQZCeOcM6HkWL4zAU4BSUUdGC5DZSoSxx0rOZsALOOGXEJI1%2B4G4z1hZqs0hJ1Z7fSafvuUtyEi0b%2BxWKXbzMy3uqVXgvXPvYr9jrJrq7%2FB1zTK5seg6TGS9YjzA2yuiPwTAQAA

### Expected behavior

qcData.flawImageList

### Actual behavior

_No response_

### Version

1.3.66

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes #7612.

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_transforms_compat

File: crates/swc/tests/fixture/issues-7xxx/7612/input/1.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+            }))}
+            showIndicators
+            useSafeAreaBottom
+            showDescription
+            current={current}
+        >
+            test
+        </Album>
+    )
+}
```
Comment (jridgewell) [inline crates/swc/tests/fixture/issues-7xxx/7612/input/1.js:16]: Nit: this is enough to trigger the bug:
```suggestion
(qcData?.flawImageList || [])?.map()
```

File: crates/swc_ecma_transforms_compat/src/es2020/optional_chaining.rs [hunk 2]
```diff
@@ -64,6 +64,7 @@ impl VisitMut for OptChaining {
             Expr::OptChain(v) => {
                 let data = self.gather(v.take(), vec![]);
                 *e = self.construct(data, false, self.c.no_document_all);
+                e.visit_mut_children_with(self);
```
Comment (jridgewell) [inline crates/swc_ecma_transforms_compat/src/es2020/optional_chaining.rs:67]: We would need the same call for `Expr::Unary`, but I think the better solution is to fix it at https://github.com/kdy1/swc/blob/9f44d10bb2a28aa7d776c2a4243074e8f59494f7/crates/swc_ecma_transforms_compat/src/es2020/optional_chaining.rs#L207. That'll avoid a repeated iteration of `prop` and `args` (we do them during `gathering`). The only thing I missed was transforming the `base`

File: crates/swc_ecma_transforms_compat/tests/optional-chaining/general-member-access/output.js [hunk 3]
```diff
@@ -8,4 +8,4 @@ var _foo, _a_b_c, _a, _a_b_c_d, _a_b, _a_b_c_d1, _a_b_c1, _orders, _orders_, _or
 orders[(_client = client) === null || _client === void 0 ? void 0 : _client.key].price;
 (_orders_client_key = orders[client.key]) === null || _orders_client_key === void 0 ? void 0 : _orders_client_key.price;
 (0, (_a1 = a) === null || _a1 === void 0 ? void 0 : _a1.b).c;
-(0, (_c = (0, a?.b).c) === null || _c === void 0 ? void 0 : _c.d).e;
```
Comment (jridgewell) [inline crates/swc_ecma_transforms_compat/tests/optional-chaining/general-member-access/output.js:11]: 🤦

**Review by jridgewell — CHANGES_REQUESTED** [review]

**Review by jridgewell — DISMISSED** [review]

**Pull request merged**